mirrord/auth/src/credential_store.rs

```diff
@@ -133,19 +131,20 @@ impl CredentialStoreSync {
 
         store.save(store_file).await?;
 
-        Ok(certificate_der)
+        result
     }
 
-    /// Get client certificate while keeping an exclusive lock on `CREDENTIALS_PATH` (and releasing
-    /// it regrading of result)
-    pub async fn get_client_certificate<R>(
+    /// Actual implementation for exclusive lock on `CREDENTIALS_PATH`.
+    pub async fn with_exclusive_lock<R, C, V>(
         client: &Client,
         credential_name: String,
-    ) -> Result<Vec<u8>>
+        callback: C,
```

Comment:
Nitpick - since we don't know what the callback is, we should probably be prepared for a panic. Right now, if the callback panics, we don't unlock the file

Reply:
Realised I don't need it anymore =]